src/zenml/models/filter_models.py

```diff
@@ -0,0 +1,383 @@
+                    ):
+                        try:
+                            value = UUID(value)
+                        except ValueError:
+                            raise ValueError(
+                                "Invalid value passed as UUID as "
+                                "query parameter."
+                            ) from 3
+                    elif operator != GenericFilterOps.EQUALS:
+                        value = str(value)
+
+                    list_of_filters.append(
+                        UUIDFilter(
+                            operation=GenericFilterOps(operator),
+                            column=key,
+                            value=value,
+                        )
+                    )
+                elif issubclass(int, get_args(cls.__fields__[key].type_)):
+                    list_of_filters.append(
+                        NumericFilter(
+                            operation=GenericFilterOps(operator),
+                            column=key,
+                            value=int(value),
+                        )
+                    )
+                elif issubclass(bool, get_args(cls.__fields__[key].type_)):
+                    list_of_filters.append(
+                        BoolFilter(
+                            operation=GenericFilterOps(operator),
+                            column=key,
+                            value=bool(value),
+                        )
+                    )
+                elif (
+                    issubclass(str, get_args(cls.__fields__[key].type_))
+                    or cls.__fields__[key].type_ == str
+                ):
+                    list_of_filters.append(
+                        StrFilter(
+                            operation=GenericFilterOps(operator),
+                            column=key,
+                            value=value,
+                        )
+                    )
+                else:
+                    logger.warning(
+                        "The Datatype "
+                        "cls.__fields__[key].type_ might "
+                        "not be supported for filtering "
+                    )
+                    list_of_filters.append(
+                        StrFilter(
+                            operation=GenericFilterOps(operator),
+                            column=key,
+                            value=str(value),
+                        )
+                    )
+
+        values["list_of_filters"] = list_of_filters
+        return values
+
+    def get_pagination_params(self) -> RawParams:
+        return RawParams(
+            limit=self.size,
+            offset=self.size * (self.page - 1),
+        )
+
+    def generate_filter(self, table: Type[SQLModel]):
+        ands = []
+        for column_filter in self.list_of_filters:
+            ands.append(column_filter.generate_query_conditions(table=table))
+
+        return ands
+
+    @classmethod
+    def click_list_options(cls):
```

Comment:
Our core models should not have any connection to `click`, I think this should rather be utility functions in `zenml/cli`.

Reply:
i can move it there, (it will make the decorator a bit more involved to use as the Filter class would need to be passed into the decorator as a parameter

Reply:
done